presto-docs/src/main/sphinx/release/release-0.193.rst

```diff
@@ -0,0 +1,76 @@
+  with multiple predicates that couldn't be pushed down to the connectors.
+* Fix a bug that may cause queries with ``row_number()`` to fail or return incorrect results.
+  If the query had an equality predicate against ``-1`` or ``0`` the bug would cause the query to fail.
+  If the query had a less-than-or-equal-to predicate against a negative value the bug would cause
+  the query to produce incorrect results.
+* Fix query analysis failure when lambda is used in ``UNNEST`` or ``VALUES``.
+* Add the peak of total non-revocable memory reservation (user + system memory) to query stats.
+* Add an optimizer rule that rewrites table scans in queries against partitioned tables to an empty relation
+  when the query has predicate(s) that do not match any partitions of the table.
+* Improve the performance of the :func:`ST_Contains` function.
+* Turn on new local scheduling algorithm by default (described in 0.181 release notes).
+* Remove the ``information_schema.__internal_partitions__`` table. This table was only useful
+  for the Hive connector and it became obsolete after ``SHOW PARTITIONS`` has been improved.
+
+Security Changes
+----------------
+
+* When using multiple authentication types the authentication methods are now applied in the order
+  they are specified in the ``http-server.authentication.type`` configuration.
+
+CLI Changes
+-----------
+
+* Fix CLI to abort the query when the pager exits.
+* Fix rendering of maps of Bing tiles in the CLI.
+
+JDBC Driver Changes
+-------------------
+
+* Use SSL by default for port 443.
+
+Hive Changes
+------------
+
+* Fix a bug that causes a ``ConcurrentModificationException`` when dropping any column
+  other than the last one.
+* Fix a bug that causes a newly added partition to have incorrect on-disk data format
+  for tables with non-default text format.
+* Enable ``SHOW PARTITIONS FROM <table> WHERE <condition>`` to work for tables
+  that have more than ``hive.max-partitions-per-scan`` partitions as long as
+  the specified ``<condition>`` reduces the number of partitions below this limit.
+* Fix a bug in ``PrestoS3FileSystem`` that causes queries to fail if they read data from
+  regions other than ``us-east-1``. Previously, such queries would fail with
+  ``"The authorization header is malformed; the region 'us-east-1' is wrong; expecting '<region_name>'"``,
+  where ``<region_name>`` is the S3 region hosting the bucket that is queried.
+
+Blackhole Changes
+-----------------
+
+* Add support for ``CREATE SCHEMA``.
+* Fix a bug that allowed creating tables in a nonexistent schema.
+
+Memory Connector Changes
+------------------------
+
+* Fix a bug in ``ALTER TABLE ... RENAME TO ...`` that was causing the source schema to be used if the
+  destination schema was different the source schema.
+* Fix a bug that allowed creating tables in a nonexistent schema.
+
+MongoDB Changes
+---------------
+
+* Fix ``INSERT INTO`` support.
```

Comment:
What was broken?

Reply:
In https://github.com/prestodb/presto/issues/9802 @kokosing says

> INSERT INTO did not work for mongodb at all, as there was no insert related method in connector, see #9736.

Reply:
Exactly. INSERT didn't work at all, since related API methods were implemented only partially (finish was not implemented).

Reply:
So then, "Add support for INSERT INTO"?

Reply:
It looks like at some point in the time it was supported, but there were no tests... and you know what happened next.

Reply:
Looking at the commits I guess `INSERT INTO` stopped working after the legacy connector API is removed in `0.155`, which was calling `commitInsert()` that the connector was already implementing. When the legacy API is removed `finishInsert()` is called by the engine, so it's probably broken since `0.155`.

I also find the current version of this release note item vague, an alternative is to give some context:

`* Fix ``INSERT INTO`` support, which was broken since the removal of the legacy connector API in :doc:`release-0.155`.`

Reply:
The command is ``INSERT``, not ``INSERT INTO`` (see it's title in the manual).

```
* Add ``INSERT`` support. It was previously removed in 0.155.